File: src/modules/health/health.controller.ts
import { Controller, Get } from "@nestjs/common";

import { Public } from "@/common/auth/public.decorator";

@Controller("health")
export class HealthController {
  @Public()
  @Get("live")
  live() {
    return {
      data: {
        status: "ok"
      }
    };
  }

  @Public()
  @Get("ready")
  ready() {
    return {
      data: {
        status: "ready"
      }
    };
  }
}

File: apps/api-java/src/main/java/com/ledgercreditsystem/api/health/HealthController.java
package com.ledgercreditsystem.api.health;

import com.ledgercreditsystem.api.common.api.StatusEnvelope;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/v1/health")
public class HealthController {

  @GetMapping("/live")
  public StatusEnvelope live() {
    return StatusEnvelope.ok("ok");
  }

  @GetMapping("/ready")
  public StatusEnvelope ready() {
    return StatusEnvelope.ok("ready");
  }
}
